build-pine64-image.sh

```diff
@@ -1,20 +1,63 @@
 #!/bin/sh
 #
-# This scripts takes a simpleimage and a kernel tarball, resizes the
+# This script takes a simpleimage and a kernel tarball, resizes the
 # secondary partition and creates a rootfs inside it. Then extracts the
 # Kernel tarball on top of it, resulting in a full Pine64 disk image.
 #
-# Latest stuff can be found at the following locations:
-# -  https://www.stdin.xyz/downloads/people/longsleep/pine64-images/simpleimage-pine64-latest.img.xz
-# -  https://www.stdin.xyz/downloads/people/longsleep/pine64-images/linux/linux-pine64-latest.tar.xz"
+# If a simpleimage is not available, the script will attempt to build one.
+
+TYPE="$1"
+LINUX="$2"
+OUTPUT="$3"
+DISTRO="$4"
+SIMPLEIMAGE="$5"
+
+SIZE="1440" # MiB
+PWD=$(pwd)
+
+set -e
+
+if [ -z "$TYPE" -o -z "$LINUX" -o -z "$OUTPUT" ]; then
+	echo "Usage: $0 <image-type> <linux-folder-or-tarball> <destination-directory> [distro] [simpleimage.img.xz]"
+	echo ""
+	echo "Image type:"
+	echo ""
+	echo "pine64      - Pine A64"
+	echo "pine64lcd   - Pine A64 with LCD"
+	echo "sopine      - SoPine A64"
+	echo "pinebook    - Pinebook (PLACEHOLDER ONLY)"
+	echo ""
+	echo "Distros available:"
+	echo ""
+	echo "arch - Arch Linux"
+	echo "xenial - Ubuntu Linux (Xenial Xerus)"
+	echo "sid - Debian Linux (sid)"
+	echo "jessie - Debian Linux (jessie)"
+	echo "opensuse - OpenSUSE Tumbleweed"
+	echo ""
+	echo "If no distro is specified, this tool will default to xenial"
+	echo "Also, if the location to the simpleimage is not provided, the tool will try"
+	echo "to build one if the required dependecies are installed"
+	exit 1
+fi
 
-SIMPLEIMAGE="$1"
-KERNELTAR="$2"
-DISTRO="$3"
-COUNT="$4"
+DISTRO2=$DISTRO
+
+if [ $TYPE = "pine64" ]; then
+	echo -n
+elif [ $TYPE = "pine64lcd" ]; then
+	DISTRO=${DISTRO}-lcd
+elif [ $TYPE = "sopine" ]; then
+	echo -n
+elif [ $TYPE = "pinebook" ]; then
+	echo -n
+else
+	echo "Invalid image type specified"
+	exit 2
+fi
 
-if [ -z "$SIMPLEIMAGE" -o -z "$KERNELTAR" ]; then
-	echo "Usage: $0 <simpleimage.img.xz> <kernel.tar.xz> [distro] [count]"
+if [ ! -d "$PWD/$OUTPUT" ]; then
```

Comment:
Please do not change usage or behavior in a way which is incompatible with how the scripts currently work.